crates/solver/src/liquidity/zeroex.rs

```diff
@@ -424,4 +615,51 @@ pub mod tests {
             .concat(),
         );
     }
+
+    #[tokio::test]
+    #[ignore]
+    async fn connect_and_update_cache_test() {
+        let cache: Arc<Mutex<HashMap<OrderId, OrderRecord>>> = Default::default();
+        let inner = cache.clone();
+        let api = Arc::new(DefaultZeroExApi::test());
+
+        let sender = {
+            let transport = create_env_test_transport();
+            let web3 = Web3::new(transport);
+            GPv2Settlement::deployed(&web3).await.unwrap().address()
+        };
+
+        tokio::task::spawn(async move {
+            let mut backoff = Duration::from_secs(3);
+
+            loop {
+                // initialize cache by fetching all existing orders via http api
+                if let Err(err) = init_cache(api.clone(), inner.clone(), sender).await {
+                    println!(
+                        "error initializing cache: {}, reconnecting with backoff {:?}",
+                        err, backoff
+                    );
+                    tokio::time::sleep(backoff).await;
+                    backoff *= 2;
+                    continue;
+                }
+                // from now on, rely on websocket connection to do incremental updates
+                if let Err(err) = connect_and_update_cache(inner.clone()).await {
+                    println!("Error updating 0x cache, reconnecting... {:?}", err);
+                    tokio::time::sleep(tokio::time::Duration::from_secs(3)).await;
+                }
+            }
+        });
```

Comment:
Shouldn't this instantiate our object under test to make sure you are actually testing the logic that is being implemented?

Reply:
Yes, created a `ZeroExCache` object.